src/node/rpc/forwarder.h

```diff
@@ -433,8 +457,19 @@ namespace ccf
             auto it = timeout_tasks.find(cmd_id);
             if (it != timeout_tasks.end())
             {
-              threading::ThreadMessaging::instance().cancel_timer_task(
-                it->second);
+              if (threading::get_current_thread_id() != it->second.thread_id)
```

Comment:
I believe `it->second.thread_id` should always match `threading::ThreadMessaging::instance().get_execution_thread(rep->client_session_id)`, and was hoping we could use that here rather than storing the thread ID. However I've spotted why that's not so easy - the `client_session_id` is in the message body rather than the header, so not trivially visible here! We could do a small refactor to make that possible (move the cancellation _after_ the calls to `recv_forwarded_response`), but I don't think its worth it.

Storing the originating thread ID like this is simpler, good spot!